xtp/src/libxtp/tools/gpu_benchmark.cc

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2009-2021 The VOTCA Development Team (http://www.votca.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Standard includes
+#include <chrono>
+#include <iomanip>
+#include <numeric>
+#include <string>
+
+#include "votca/tools/property.h"
+#include "votca/xtp/bse_operator.h"
+#include "votca/xtp/eigen.h"
+#include "votca/xtp/logger.h"
+#include "votca/xtp/openmp_cuda.h"
+#include "votca/xtp/orbitals.h"
+#include "votca/xtp/rpa.h"
+#include "votca/xtp/threecenter.h"
+
+// Local private VOTCA includes
+#include "gpu_benchmark.h"
+
+namespace votca {
+namespace xtp {
+
+void GPUBenchmark::ParseOptions(const tools::Property& options) {
+
+  repetitions_ = options.get("repetitions").as<Index>();
+}
+
+std::pair<double, double> CalcStatistics(const std::vector<double>& v) {
+  double mean = std::reduce(v.begin(), v.end()) / ((double)v.size());
+  double sq_sum = std::inner_product(v.begin(), v.end(), v.begin(), 0.0);
+  double stdev = std::sqrt(sq_sum / ((double)v.size()) - mean * mean);
+  return {mean, stdev};
+}
+
+template <class T>
+tools::Property RunPart(T&& payload, const std::string& name,
+                        Index repetitions) {
+  std::vector<double> individual_timings;
+  individual_timings.reserve(repetitions);
+  Index count = 0;
+  std::cout << name << std::endl;
+  for (Index i = 0; i < repetitions; i++) {
+    std::chrono::time_point<std::chrono::steady_clock> start =
+        std::chrono::steady_clock::now();
+    count += payload();
```

Comment:
@JensWehner codacy thinks `count` is never used, please have a look.

Reply:
should be done now